### PR TITLE
added withSuitableRoot() extension

### PR DESCRIPTION
**Pull request by itsergpot**

### CODE REVIEW

File: ultron/src/main/java/com/atiurin/ultron/custom/espresso/matcher/SuitableRootMatcher.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.Root
+import com.atiurin.ultron.core.espresso.UltronEspressoInteraction
+import com.atiurin.ultron.core.espresso.recyclerview.UltronRecyclerView
+import com.atiurin.ultron.extensions.withSuitableRoot
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+
+fun withSuitableRoot(viewMatcher: Matcher<View>): Matcher<Root> {
```
Comment (alex-tiurin) [inline ultron/src/main/java/com/atiurin/ultron/custom/espresso/matcher/SuitableRootMatcher.kt:15]: Pls, add a javadoc with description of the method purpose. User should understand the usecase

File: ultron/src/main/java/com/atiurin/ultron/core/espresso/recyclerview/UltronRecyclerViewItem.kt [hunk 2]
```diff
@@ -77,6 +78,9 @@ open class UltronRecyclerViewItem {
     fun withAssertion(name: String = "", isListened: Boolean = false, block: () -> Unit) =
         getMatcher().withAssertion(DefaultOperationAssertion(name, block.setListenersState(isListened)))
 
+    //root view searching
```
Comment (alex-tiurin) [inline ultron/src/main/java/com/atiurin/ultron/core/espresso/recyclerview/UltronRecyclerViewItem.kt:81]: Pls, add a javadoc with description of the method purpose. User should understand the usecase

File: ultron/src/main/java/com/atiurin/ultron/extensions/ViewInteractionExt.kt [hunk 3]
```diff
@@ -20,6 +24,17 @@ fun ViewInteraction.withAssertion(assertion: OperationAssertion) = UltronEspress
 fun ViewInteraction.withAssertion(name: String = "", isListened: Boolean = false, block: () -> Unit) =
     UltronEspressoInteraction(this).withAssertion(DefaultOperationAssertion(name, block.setListenersState(isListened)))
 
+//root view searching
```
Comment (alex-tiurin) [inline ultron/src/main/java/com/atiurin/ultron/extensions/ViewInteractionExt.kt:27]: Pls, add a javadoc with description of the method purpose. User should understand the usecase

**Review by alex-tiurin — CHANGES_REQUESTED** [review]

File: ultron/src/main/java/com/atiurin/ultron/core/espresso/recyclerview/UltronRecyclerViewItem.kt (hunk 2 above)
Comment (itsergpot) [reply ultron/src/main/java/com/atiurin/ultron/core/espresso/recyclerview/UltronRecyclerViewItem.kt:81]: https://github.com/open-tool/ultron/wiki/Avoiding-nontrivial-element-lookup-exceptions

File: ultron/src/main/java/com/atiurin/ultron/custom/espresso/matcher/SuitableRootMatcher.kt (hunk 1 above)
Comment (itsergpot) [reply ultron/src/main/java/com/atiurin/ultron/custom/espresso/matcher/SuitableRootMatcher.kt:15]: https://github.com/open-tool/ultron/wiki/Avoiding-nontrivial-element-lookup-exceptions

File: ultron/src/main/java/com/atiurin/ultron/extensions/ViewInteractionExt.kt (hunk 3 above)
Comment (itsergpot) [reply ultron/src/main/java/com/atiurin/ultron/extensions/ViewInteractionExt.kt:27]: https://github.com/open-tool/ultron/wiki/Avoiding-nontrivial-element-lookup-exceptions

**Pull request merged**